### PR TITLE
Add ThreadConfigSetter for Embedder api that can set thread priority

### PR DESCRIPTION
**Pull request by JsouLiang**
Change the `embedder.cc` create thread host api for that we can specified the thread priority

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by JsouLiang** [comment]
@dnfield hey, please give me a review?

**Review by dnfield — COMMENTED** [review]
This seems right to me - please add a test to embedder_unittests.cc for this

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by dnfield** [comment]
LGTM once tests are passing anyway :)

**Comment by JsouLiang** [comment]
> LGTM once tests are passing anyway :)

testing done

**Comment by dnfield** [comment]
This needs a second review, probably from @chinmaygarde or @gaaclarke

**Comment by dnfield** [comment]
@cbracken maybe?

**Comment by gaaclarke** [comment]
I reviewed the changes for iOS and Android, this LGTM.

**Pull request merged**
